### ISSUE TITLE
Fix abjad.mutate().split() error

### ISSUE THREAD

**Issue by trevorbaca** [Abjad/abjad#272]
_From [josiah.oberholtzer](https://code.google.com/u/josiah.oberholtzer/) on October 19, 2013 11:35:53_

Splitting errors when trying to split a note inside a tuplet, which then results in child tuplets containing the split leaf parts, which are then to be tied together.

Here, the e'16 is to be split:
```
>>> voice = Voice(r"c'4 \times 3/4 { d'32 e'16. f'8 }")
>>> mutate(voice[:]).split([(5, 16)])
```
And output:
```
Traceback (most recent call last):
  File "split_error.py", line 4, in <module>
    mutate(voice[:]).split([(5, 16)])
  File "/Users/josiah/Documents/Development/abjadroot/trunk/abjad/tools/mutationtools/ScoreMutationAgent/ScoreMutationAgent.py", line 1392, in split
    tie_split_notes=tie_split_notes, 
  File "/Users/josiah/Documents/Development/abjadroot/trunk/abjad/tools/containertools/Container/Container.py", line 783, in _split_by_duration
    tie_split_notes=tie_split_notes,
  File "/Users/josiah/Documents/Development/abjadroot/trunk/abjad/tools/leaftools/Leaf/Leaf.py", line 557, in _split_by_duration
    selection._attach_tie_spanner_to_leaf_pair()
  File "/Users/josiah/Documents/Development/abjadroot/trunk/abjad/tools/selectiontools/ContiguousSelection/ContiguousSelection.py", line 63, in _attach_tie_spanner_to_leaf_pair
    assert isinstance(left_leaf, leaftools.Leaf), left_leaf
AssertionError: {\* 3:2 c'16, c'64 *}
```
_Original issue: http://code.google.com/p/abjad/issues/detail?id=272_

**Comment by trevorbaca** [comment]
Still valid in June 2017:
```
voice = Voice(r"c'4 \times 3/4 { d'32 e'16. f'8 }")
mutate(voice[:]).split([(5, 16)])
```
Output:
```
Traceback (most recent call last):
  File "/Users/trevorbaca/.abjad/abjad_tmp.py", line 9, in <module>
    mutate(voice[:]).split([(5, 16)])
  File "/Users/trevorbaca/abjad/abjad/tools/agenttools/MutationAgent.py", line 2660, in split
    use_messiaen_style_ties=use_messiaen_style_ties,
  File "/Users/trevorbaca/abjad/abjad/tools/scoretools/Container.py", line 1036, in _split_by_duration
    use_messiaen_style_ties=use_messiaen_style_ties,
  File "/Users/trevorbaca/abjad/abjad/tools/scoretools/Leaf.py", line 550, in _split_by_duration
    use_messiaen_style_ties=use_messiaen_style_ties,
  File "/Users/trevorbaca/abjad/abjad/tools/selectiontools/Selection.py", line 374, in _attach_tie_spanner_to_leaf_pair
    assert isinstance(left_leaf, scoretools.Leaf), left_leaf
AssertionError: Tuplet(Multiplier(2, 3), "c'16 ~ c'64")
```

**Comment by trevorbaca** [comment]
Simplification:
```
>>> staff = abjad.Staff(r"\time 7/16 c'4 \times 3/4 { d'32 e'16. f'8 }")
>>> leaf = abjad.inspect(staff).get_leaf(2)
>>> leaf._split_by_duration((5, 128))

Traceback (most recent call last):
  File "/Users/trevorbaca/.abjad/abjad_tmp.py", line 11, in <module>
    leaf._split_by_duration((5, 128))
  File "/Users/trevorbaca/abjad/abjad/tools/scoretools/Leaf.py", line 561, in _split_by_duration
    use_messiaen_style_ties=use_messiaen_style_ties,
  File "/Users/trevorbaca/abjad/abjad/tools/selectiontools/Selection.py", line 390, in _attach_tie_spanner_to_leaf_pair
    assert isinstance(left_leaf, abjad.Leaf), left_leaf
AssertionError: Tuplet(Multiplier(2, 3), "e'16 ~ e'64")
```

**Comment by trevorbaca** [comment]
Fixed in 2.21:
```
>>> staff = abjad.Staff(r"\time 7/16 c'4 \times 3/4 { d'32 e'16. f'8 }")
>>> leaf = abjad.inspect(staff).get_leaf(2)
>>> leaf._split_by_duration((5, 128))

>>> f(staff)
\new Staff {
    \time 7/16
    c'4
    \tweak text #tuplet-number::calc-fraction-text
    \times 3/4 {
        d'32
        \times 2/3 {
            e'16 ~
            e'64 ~
            e'16
        }
        f'8
    }
}
```
